File: frontend/types/report.ts
// Tipos para reportes operativos (CU-22) y reportes por voz (CU-37).

export type ReportFormat = 'pdf' | 'xlsx'

export interface ReportFilters {
  format: ReportFormat
  from?: string
  to?: string
  status?: string
  analyst?: string
  supervisor?: string
  policyholder?: string
  q?: string
}

// ── CU-37: interpretación de un pedido de reporte por voz ──────────────

export interface VoiceReportResolved {
  format: string
  status_label?: string | null
  analyst_label?: string | null
  supervisor_label?: string | null
  policyholder_label?: string | null
}

export interface VoiceReportInterpretation {
  transcript: string
  supported: boolean
  note: string
  filters: {
    format: ReportFormat
    status?: string | null
    from?: string | null
    to?: string | null
    analyst?: string | null
    supervisor?: string | null
    policyholder?: string | null
    q?: string | null
  }
  resolved: VoiceReportResolved
  warnings: string[]
}

File: backend/app/schemas/report.py
"""Schemas de reportes por voz — CU-37."""

from pydantic import BaseModel, Field


class ReportInterpretRequest(BaseModel):
    """Texto ya transcripto (fallback / tests sin audio)."""

    text: str = Field(min_length=1, max_length=1000)


class VoiceReportFilters(BaseModel):
    """Filtros listos para alimentar `GET /api/reports/claims`."""

    format: str = "pdf"
    status: str | None = None
    from_: str | None = Field(default=None, alias="from")
    to: str | None = None
    analyst: str | None = None
    supervisor: str | None = None
    policyholder: str | None = None
    q: str | None = None

    model_config = {"populate_by_name": True}


class VoiceReportResolved(BaseModel):
    """Etiquetas legibles para la pantalla de confirmación."""

    format: str
    status_label: str | None = None
    analyst_label: str | None = None
    supervisor_label: str | None = None
    policyholder_label: str | None = None


class VoiceReportInterpretation(BaseModel):
    """Resultado de interpretar un pedido de reporte (texto o voz)."""

    transcript: str
    supported: bool
    note: str = ""
    filters: VoiceReportFilters
    resolved: VoiceReportResolved
    warnings: list[str] = []
